CondFormats/HcalObjects/interface/HcalQIECoderExtended.h

```diff
@@ -0,0 +1,92 @@
+#ifndef HcalQIECoderExtended_h
+#define HcalQIECoderExtended_h
+
+/** 
+\class HcalQIECoderExtended
+\author  Clemencia Mora (CBPF)
+Extended version of QIECoder to add QIEId: barcode and channel
+$Author: cmora
+$Date: 2015/01/13  $
+$Revision: 1.0 $
+*/
+
+#include "CondFormats/Serialization/interface/Serializable.h"
+//#include "CondFormats/HcalObjects/interface/HcalQIECoder.h"
+
+#include <boost/cstdint.hpp>
+
+#include <vector>
+#include <algorithm>
+
+class HcalQIEShape;
+
+class HcalQIECoderExtended {
+ public:
+  HcalQIECoderExtended (unsigned long fId = 0) : mId(fId), mQIEbarcode(0), mQIEchannel(0) { setQIEIndex(0);}
```

Comment:
@clemencia 
* avoid usage of `unsigned long`: it can be either 32 or 64 bits according to the arch;
* `fId` is used to initialise a 32 bits value: so, either you make it `uint32_t` or you make both `unsigned long long`.